### ISSUE TITLE
Bug on line 312 of @sdpvar/times.m?

### ISSUE THREAD

**Issue by giofantuzzi** [yalmip/YALMIP#460]
Dear Johan,

there seems to a bug on [line 312 of `@sdpvar/times.m`](https://github.com/yalmip/YALMIP/blob/develop/%40sdpvar/times.m#L307-312). I am using YALMIP version 20171018 after pullling the `master` branch from the GitHub repo on 27 Oct 2017. Typing

```
>> yalmip clear
>> x = sdpvar(3);
>> 3.*x
```
returns the error
```
Error using sdpvar/reshape (line 27)
Error using reshape
To RESHAPE the number of elements must not change.

Error in  .*  (line 312)
y = reshape(y,size(X));
```
The same issue arises when multiplying a numeric scalar by `sdpvar` column/row vectors. Before updating to the last version of YALMIP the issue did not occur (working code using `.*` failed to run after the update). The problem does not appear when a numeric array is multiplied by a scalar `sdpvar`:
```
>> yalmip clear
>> sdpvar x
>> sdisplay(x.*rand(3))

ans =

  3×3 cell array

    '0.814723686393*x'    '0.913375856139*x'     '0.278498218867*x'
    '0.905791937076*x'    '0.632359246225*x'     '0.546881519205*x'
    '0.126986816294*x'    '0.0975404049994*x'    '0.957506835434*x'
```
Of course I can replace `.*` with `*` everywhere, but fixing the issue would make YALMIP more robust. The change suggested below fixed the multiplication problem, but creates a small problem with `sdisplay`. Would it be possible to look into this?

Thanks in advance,
Giovanni

**Suggested fix:** replace `y = reshape(y,size(X));` on [line 312 of `@sdpvar/times.m`](https://github.com/yalmip/YALMIP/blob/develop/%40sdpvar/times.m#L307-312) with
```
if numel(y)==numel(X)
    y = reshape(y,size(X));
else
    y = reshape(y,size(Y));
end
```
Then
```
>> yalmip clear
>> x = sdpvar(3);
>> sdisplay(3.*x)

ans =

  3×3 cell array

    '3*x(1,1)'    '3*x(2,1)'    '3*x(3,1)'
    '3*x(2,1)'    '3*x(2,2)'    '3*x(3,2)'
    '3*x(3,1)'    '3*x(3,2)'    '3*x(3,3)'
```
However,
```
>> 3.*x
Linear matrix variable 3x3 (symmetric, real, 6 variables)
>> sdisplay(ans)
Undefined function or variable 'names'.

Error in sdisplay (line 140)
                        already = ~strfind(names{index_in_p},'internal');
```

**Comment by johanlofberg** [comment]
Hmm, I seem to have messed up something when I fixed the related #441. Will fix of course

**Comment by giofantuzzi** [comment]
Thanks!
